openupgrade_scripts/scripts/project/15.0.1.2/pre-migration.py

```diff
@@ -71,10 +71,56 @@ def rename_project_milestone_target_date(env):
         )
 
 
+def migrate_project_status(env):
+    """Migrate project_status module if installed to
+    project.project.stages.
+    """
+    if openupgrade.table_exists(env.cr, "project_status"):
+        openupgrade.rename_models(
+            env.cr, [("project.status", "project.project.stage")]
+        )
+        openupgrade.rename_tables(
+            env.cr, [("project_status", "project_project_stage")]
+        )
+        openupgrade.rename_fields(
+            env,
+            [
+                (
+                    "project.project.stage",
+                    "project_project_stage",
+                    "status_sequence",
+                    "sequence",
+                ),
+                (
+                    "project.project",
+                    "project_project",
+                    "project_status",
+                    "stage_id",
+                ),
+            ]
+        )
+        openupgrade.rename_columns(
+            env.cr,
+            {
+                "project_project_stage": [
+                    ("company_id", None),
+                    ("description", None),
+                    ("is_closed", None),
+                ],
+            },
+        )
+        # We need to delete old project_status views
+        openupgrade.logged_query(
+            env.cr,
+            "DELETE FROM ir_ui_view WHERE model = 'project.project';"
+        )
```

Comment:
This part may be a bit overkill. But I need to delete `project_status` views because there is references on old project.status model in it, and this causes the database to fail showing project views.

But views where renamed during updade of the base module, so I didn't find a way to distinguish between views that are part of existing project module and views coming from project_status and that were renamed.

I tried to remove old views by xmlids but this does not work. I think that this is because some xmlids ends up being the same after the renaming (e.g. `project_status.edit_project` and `project.edit_project`).

By using this, I force the update of project module to recreate views of the project.project model.

Any better way of doing it is welcome.

Reply:
Indeed this is not OK. Regular update process (the second pass without OpenUpgrade) must remove them being noupdate="0".

Reply:
I tried not to delete `project.project` views, and perform a regular update after migration in 15.0.

I got the following logs:

```
2024-04-23 08:39:02,435 7 INFO brupower_upgrade odoo.addons.base.models.ir_module: ALLOW access to module.button_immediate_upgrade on ['Projet'] to user __system__ #1 via n/a 
2024-04-23 08:39:02,436 7 INFO brupower_upgrade odoo.addons.base.models.ir_module: ALLOW access to module.button_upgrade on ['Projet'] to user __system__ #1 via n/a 
2024-04-23 08:39:02,436 7 INFO brupower_upgrade odoo.addons.base.models.ir_module: ALLOW access to module.update_list on ['Projet'] to user __system__ #1 via n/a 
2024-04-23 08:39:09,805 7 INFO brupower_upgrade odoo.addons.base.models.ir_module: ALLOW access to module.button_install on [] to user __system__ #1 via n/a 
...
2024-04-23 08:39:19,078 7 INFO brupower_upgrade odoo.modules.loading: loading project/views/project_views.xml 
2024-04-23 08:39:19,253 7 WARNING brupower_upgrade odoo.addons.openupgrade_framework.odoo_patch.odoo.addons.base.models.ir_ui_view: Can't render custom view project.project_view_kanban for model project.project. Assuming you are migrating 
between major versions of Odoo. Please review the view contents manually after the migration.
Error: Element '<kanban class="o_kanban_mobile">' cannot be located in parent view 
2024-04-23 08:39:19,286 7 WARNING brupower_upgrade odoo.addons.openupgrade_framework.odoo_patch.odoo.addons.base.models.ir_ui_view: Can't render custom view project.project_view_kanban for model project.project. Assuming you are migrating 
between major versions of Odoo. Please review the view contents manually after the migration.
Error: Element '<kanban class="o_kanban_mobile">' cannot be located in parent view 
2024-04-23 08:39:19,325 7 INFO brupower_upgrade odoo.models.unlink: User #1 deleted ir.actions.act_window.view records with IDs: [336, 337] 
2024-04-23 08:39:19,593 7 WARNING brupower_upgrade odoo.addons.openupgrade_framework.odoo_patch.odoo.addons.base.models.ir_ui_view: Can't render custom view project.project_view_kanban for model project.project. Assuming you are migrating 
between major versions of Odoo. Please review the view contents manually after the migration.
Error: Element '<kanban class="o_kanban_mobile">' cannot be located in parent view 
2024-04-23 08:39:19,598 7 INFO brupower_upgrade odoo.modules.loading: loading project/views/res_partner_views.xml 
...
```

So it seams that I need to delete some views, but may be not all.

Reply:
That's not a regular update, as `openupgrade_framework` is still around. Regular update will remove them.

Reply:
Ho, yes indeed.

I've tested to deactivate views instead of deleting it, and it works also.

Reply:
To perform a regular update after this migration in 15.0, I need to add a step to my migration process. Without deactivating views, if I go directly from a upgrade in 15.0 with openupgrade to an upgrade in 16.0 with openupgrade then upgrade in 16.0 with openupgrade fails.

Reply:
Well, that shouldn't happen. It will emit the same warnings, but no crash.

Reply:
I fixed it by just removing views that were overridden in the `project_status` module. And it works well.
Some standard views of `project` module were deleted in `pre-migration` script and then created fresh new by the update.